internal/grpc/services/gateway/storageprovider.go

```diff
@@ -1089,8 +1089,7 @@ func (s *svc) getStorageProviderClient(_ context.Context, p *registry.ProviderIn
 		return nil, err
 	}
 
-	// TODO: reinstate cache as soon there is a implementation for a non-in-memory cache
-	return c, nil
+	return s.cache.StorageProviderClient(c), nil
```

Comment:
we are using a noop cache by default and ttls default to 0, also disabling the cache. We need a shared cache to make the dataprovider invalidate the stat cache when a file is written.